### PR TITLE
1612 refactor moderator spec

### PR DESCRIPTION
**Pull request by aonomike**
## 🍰 Pullrequest

Refactor tests moderation tests to use `apollo-server-testing`
### Issues

- relates #1612 

### Todo

- [X] Fix tests for moderating posts (enable)
- [X] Fix tests for enable
- [X] Remove commented code once this is complete

### CODE REVIEW

File: backend/src/schema/resolvers/moderation.spec.js [hunk 1]
```diff
@@ -54,369 +63,369 @@ describe('disable', () => {
...
-      })
+  describe('unauthenticated', () => {
+    beforeEach(async () => {
+      authenticatedUser = await currentUser.toJson()
     })
-
     it('throws authorization error', async () => {
-      await setup()
-      await expect(action()).rejects.toThrow('Not Authorised')
+      const { errors } = await mutate({ mutation, variables })
```
Comment (roschaefer) [inline backend/src/schema/resolvers/moderation.spec.js]:
```suggestion
      await expect(mutate({ mutation, variables })).resolves.toMatchObject({
        errors: [{message: 'Not Authorised!'}]
      })
```

File: backend/src/schema/resolvers/moderation.spec.js [hunk 2]
```diff
@@ -54,369 +63,369 @@ describe('disable', () => {
...
-  })
-
-  describe('authenticated', () => {
-    beforeEach(() => {
-      authenticateClient = setupAuthenticateClient({
-        email: 'user@example.org',
-        password: '1234',
+      it('throws authorization error', async () => {
+        const { errors } = await mutate({ mutation, variables })
+        expect(errors[0]).toHaveProperty('message', 'Not Authorised!')
```
Comment (roschaefer) [inline backend/src/schema/resolvers/moderation.spec.js]: See my suggestion above :point_up:

File: backend/src/schema/resolvers/moderation.spec.js [hunk 3]
```diff
@@ -54,369 +63,369 @@ describe('disable', () => {
...
           }
+          await Promise.all([factory.create('Tag', { id: 'sample-tag-id' })])
         })
 
         it('returns null', async () => {
-          const expected = { enable: null }
-          await setup()
-          await expect(action()).resolves.toEqual(expected)
+          const expected = { data: { disable: null } }
+          await expect(mutate({ mutation, variables })).resolves.toMatchObject(expected)
```
Comment (roschaefer) [inline backend/src/schema/resolvers/moderation.spec.js]: Not too much code here, so you could merge these two lines,

File: backend/src/schema/resolvers/moderation.spec.js [hunk 4]
```diff
@@ -54,369 +63,369 @@ describe('disable', () => {
...
+          const before = { data: { Comment: [{ id: 'comment-id', disabledBy: null }] } }
+          const expected = {
+            data: { Comment: [{ id: 'comment-id', disabledBy: { id: 'moderator-id' } }] },
           }
+          variables = { id: 'comment-id' }
+          await expect(query({ query: commentQuery, variables })).resolves.toMatchObject(before)
+          await expect(mutate({ mutation, variables })).resolves.toMatchObject({
+            data: { disable: 'comment-id' },
+          })
+          await expect(query({ query: commentQuery, variables })).resolves.toMatchObject(expected)
```
Comment (roschaefer) [inline backend/src/schema/resolvers/moderation.spec.js]: Before and each, excellent!

File: backend/src/schema/resolvers/moderation.spec.js (hunk 1 above)
Comment (aonomike) [reply backend/src/schema/resolvers/moderation.spec.js]: Done

File: backend/src/schema/resolvers/moderation.spec.js (hunk 2 above)
Comment (aonomike) [reply backend/src/schema/resolvers/moderation.spec.js]: Done

File: backend/src/schema/resolvers/moderation.spec.js (hunk 3 above)
Comment (aonomike) [reply backend/src/schema/resolvers/moderation.spec.js]: Done

**Comment by roschaefer** [comment]
@aonomike awesome. The build is failing for unused import statements (run `yarn run lint --fix`) to find them.

Do you need help to finish the `enabled` scope?

**Comment by aonomike** [comment]
> @aonomike awesome. The build is failing for unused import statements (run `yarn run lint --fix`) to find them.
> 
> Do you need help to finish the `enabled` scope?

Yes @roschaefer I am currently working on the enabled. Will probably be done later today. I will fix the lint as well!

**Pull request merged**